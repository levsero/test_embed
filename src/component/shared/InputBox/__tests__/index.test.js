import { render, fireEvent } from '@testing-library/react'
import React from 'react'

import { InputBox } from '../index'

const renderComponent = (props = {}) => {
  const defaultProps = {
    name: 'inputBox',
    placeholder: 'placeholder',
    handleSendInputValue: noop
  }

  const componentProps = {
    ...defaultProps,
    ...props
  }

  return render(<InputBox {...componentProps} />)
}

test('renders the expected classes', () => {
  const { container } = renderComponent()

  expect(container).toMatchSnapshot()
})

test('renders as disabled', () => {
  const { container } = renderComponent({
    disabled: true
  })

  expect(container).toMatchSnapshot()
})

test('renders as mobile', () => {
  const { container } = renderComponent({
    isMobile: true
  })

  expect(container).toMatchSnapshot()
})

test('renders with value', () => {
  const { container } = renderComponent({
    inputValue: 'doremi'
  })

  expect(container).toMatchSnapshot()
})

describe('events', () => {
  it('updates handler on change', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      updateInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.change(getByPlaceholderText('here'), { target: { value: 'new' } })
    expect(handler).toHaveBeenCalledWith('new')
  })

  it('updates handler on enter key event', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      handleSendInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.keyDown(getByPlaceholderText('here'), {
      key: 'Enter',
      keyCode: 13,
      which: 13
    })
    expect(handler).toHaveBeenCalled()
  })

  it('does not update handler on enter key event when shift key is pressed', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      handleSendInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.keyDown(getByPlaceholderText('here'), {
      key: 'Enter',
      keyCode: 13,
      which: 13,
      shiftKey: true
    })
    expect(handler).not.toHaveBeenCalled()
  })
})
