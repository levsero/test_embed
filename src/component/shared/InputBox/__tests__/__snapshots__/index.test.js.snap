// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as disabled 1`] = `
<div>
  <div
    class="textField fieldDisabled TextGroup-sc-127gh0a-0 text__c-txt___14ptu fjQWCj"
    data-garden-id="textfields.text_group"
    data-garden-version="3.7.0"
  >
    <label
      class="label Label-sc-129v0bx-0 text__c-txt__label___11SEJ jcOpUC"
      data-garden-id="textfields.label"
      data-garden-version="3.7.0"
      for="garden-text-field-2--input"
      id="garden-text-field-2--label"
    >
      placeholder
    </label>
    <textarea
      aria-invalid="false"
      aria-labelledby="garden-text-field-2--label"
      class="input Textarea-sc-1alsvvb-0 text__c-txt__input--area___3VBq- dcOmlb Input-sc-4gf5js-0-textarea text__c-txt__input___2ZeOl text__is-disabled___3Uekk kpjxwv"
      data-garden-id="textfields.textarea"
      data-garden-version="3.7.0"
      data-testid="message-field"
      disabled=""
      id="garden-text-field-2--input"
      name="inputBox"
      placeholder="placeholder"
      rows="3"
    />
  </div>
</div>
`;

exports[`renders as mobile 1`] = `
<div>
  <div
    class="textField TextGroup-sc-127gh0a-0 text__c-txt___14ptu fjQWCj"
    data-garden-id="textfields.text_group"
    data-garden-version="3.7.0"
  >
    <label
      class="label Label-sc-129v0bx-0 text__c-txt__label___11SEJ jcOpUC"
      data-garden-id="textfields.label"
      data-garden-version="3.7.0"
      for="garden-text-field-4--input"
      id="garden-text-field-4--label"
    >
      placeholder
    </label>
    <textarea
      aria-invalid="false"
      aria-labelledby="garden-text-field-4--label"
      class="input inputMobile fieldMobile Textarea-sc-1alsvvb-0 text__c-txt__input--area___3VBq- dcOmlb Input-sc-4gf5js-0-textarea text__c-txt__input___2ZeOl kpjxwv"
      data-garden-id="textfields.textarea"
      data-garden-version="3.7.0"
      data-testid="message-field"
      id="garden-text-field-4--input"
      name="inputBox"
      placeholder="placeholder"
      rows="1"
    />
  </div>
</div>
`;

exports[`renders the expected classes 1`] = `
<div>
  <div
    class="textField TextGroup-sc-127gh0a-0 text__c-txt___14ptu fjQWCj"
    data-garden-id="textfields.text_group"
    data-garden-version="3.7.0"
  >
    <label
      class="label Label-sc-129v0bx-0 text__c-txt__label___11SEJ jcOpUC"
      data-garden-id="textfields.label"
      data-garden-version="3.7.0"
      for="garden-text-field-0--input"
      id="garden-text-field-0--label"
    >
      placeholder
    </label>
    <textarea
      aria-invalid="false"
      aria-labelledby="garden-text-field-0--label"
      class="input Textarea-sc-1alsvvb-0 text__c-txt__input--area___3VBq- dcOmlb Input-sc-4gf5js-0-textarea text__c-txt__input___2ZeOl kpjxwv"
      data-garden-id="textfields.textarea"
      data-garden-version="3.7.0"
      data-testid="message-field"
      id="garden-text-field-0--input"
      name="inputBox"
      placeholder="placeholder"
      rows="3"
    />
  </div>
</div>
`;

exports[`renders with value 1`] = `
<div>
  <div
    class="textField TextGroup-sc-127gh0a-0 text__c-txt___14ptu fjQWCj"
    data-garden-id="textfields.text_group"
    data-garden-version="3.7.0"
  >
    <label
      class="label Label-sc-129v0bx-0 text__c-txt__label___11SEJ jcOpUC"
      data-garden-id="textfields.label"
      data-garden-version="3.7.0"
      for="garden-text-field-6--input"
      id="garden-text-field-6--label"
    >
      placeholder
    </label>
    <textarea
      aria-invalid="false"
      aria-labelledby="garden-text-field-6--label"
      class="input Textarea-sc-1alsvvb-0 text__c-txt__input--area___3VBq- dcOmlb Input-sc-4gf5js-0-textarea text__c-txt__input___2ZeOl kpjxwv"
      data-garden-id="textfields.textarea"
      data-garden-version="3.7.0"
      data-testid="message-field"
      id="garden-text-field-6--input"
      name="inputBox"
      placeholder="placeholder"
      rows="3"
    >
      doremi
    </textarea>
  </div>
</div>
`;
